packages/core/src/fhir-deduplication/compute-resources-xor.ts

```diff
@@ -0,0 +1,112 @@
+import { Resource } from "@medplum/fhirtypes";
+import { BadRequestError } from "@metriport/shared";
+import { z } from "zod";
+import { deduplicateResources } from "./dedup-resources";
+import { artifactRelatedArtifactUrl } from "./shared";
+
+/**
+ * Computes the XOR of two lists of resources of the same resource type.
+ *
+ * Important! The input resources must all be of the same resource type and have no overlapping IDs.
+ *
+ * @param targetResources - The target resources.
+ * @param sourceResources - The source resources.
+ * @param schema - The schema to parse the returned Resource objects to the input type. Optional, defaults to casting the Resource type to the input type.
+ * @returns The XOR of the target and source resources. Only resources with IDs are returned.
+ */
+export function computeResourcesXorAlongResourceType<T extends Resource>({
+  targetResources,
+  sourceResources,
+  schema,
+}: {
+  targetResources: T[];
+  sourceResources: T[];
+  schema?: z.ZodSchema<T>;
+}): {
+  computedXorTargetResources: T[];
+  computedXorSourceResources: T[];
```

Comment:
nit: maybe simplify the naming?

```ts
  targetOnly: T[];
  sourceOnly: T[];
```

Or use [Lodash's approach](https://lodash.com/docs/4.17.15#partition), returning an array of arrays, since `source` and `target` are not obvious as to what those mean.

Something like 

```ts
      const [metriportOnly, ehrOnly] = computeResourcesXorAlongResourceType([
        metriportResources,
        dedupedEhrResources,
      ]);
```

But NABD if you prefer to leave as is, just thought it was simpler.